src/connector/src/sink/iceberg/log_sink.rs

```diff
@@ -0,0 +1,141 @@
+            Uninitialized,
+
+            /// Mark that a new epoch has begun.
+            EpochBegun { curr_epoch: u64 },
+
+            /// Mark that the consumer has just received a barrier
+            BarrierReceived { prev_epoch: u64 },
+        }
+
+        let mut state = LogConsumerState::Uninitialized;
+
+        let mut current_checkpoint: u64 = 0;
+        let commit_checkpoint_interval = self.commit_checkpoint_interval;
+
+        loop {
+            let (epoch, item): (u64, LogStoreReadItem) = log_reader.next_item().await?;
+            if let LogStoreReadItem::UpdateVnodeBitmap(_) = &item {
+                match &state {
+                    LogConsumerState::BarrierReceived { .. } => {}
+                    _ => unreachable!(
+                        "update vnode bitmap can be accepted only right after \
+                    barrier, but current state is {:?}",
+                        state
+                    ),
+                }
+            }
+            // begin_epoch when not previously began
+            state = match state {
+                LogConsumerState::Uninitialized => {
+                    sink_writer.begin_epoch(epoch).await?;
+                    LogConsumerState::EpochBegun { curr_epoch: epoch }
+                }
+                LogConsumerState::EpochBegun { curr_epoch } => {
+                    assert!(
+                        epoch >= curr_epoch,
+                        "new epoch {} should not be below the current epoch {}",
+                        epoch,
+                        curr_epoch
+                    );
+                    LogConsumerState::EpochBegun { curr_epoch: epoch }
+                }
+                LogConsumerState::BarrierReceived { prev_epoch } => {
+                    assert!(
+                        epoch > prev_epoch,
+                        "new epoch {} should be greater than prev epoch {}",
+                        epoch,
+                        prev_epoch
+                    );
+                    sink_writer.begin_epoch(epoch).await?;
+                    LogConsumerState::EpochBegun { curr_epoch: epoch }
+                }
+            };
+            match item {
+                LogStoreReadItem::StreamChunk { chunk, .. } => {
+                    if let Err(e) = sink_writer.write_batch(chunk).await {
+                        sink_writer.abort().await?;
+                        return Err(e);
+                    }
+                }
+                LogStoreReadItem::Barrier { is_checkpoint } => {
+                    let prev_epoch = match state {
+                        LogConsumerState::EpochBegun { curr_epoch } => curr_epoch,
+                        _ => unreachable!("epoch must have begun before handling barrier"),
+                    };
+                    if is_checkpoint {
+                        current_checkpoint += 1;
+                        if commit_checkpoint_interval <= 1
```

Comment:
Can we just assert that the `commit_checkpoint_interval` must not be 0 and only check `current_checkpoint >= commit_checkpoint_interval` ? Currently we use `unwrap_or_default` to get the `commit_checkpoint`, which will pass 0 when not set. If we use 1 as the default value, we can avoid passing the 0 value.

Reply:
But we can't prevent the user from setting it as 0 so it means that we should adjust it to 1 if the user sets it to 0. Another concern is that this assumption is not type-safe. We can't prevent others follow this assumption(It's easy to miss), even for now, seems it will not cause something wrong. 🤔

Reply:
I checked the config and used `NonZero` internally to guarantee that the interval is bigger than 0 now.